libs/mock-block-dock/src/mock-block-dock.tsx

```diff
@@ -254,6 +275,23 @@ export const MockBlockDock: FunctionComponent<MockBlockDockProps> = ({
     valueName: "readonly",
   });
 
+  useEffect(() => {
+    if (serviceModule && serviceModuleCallbacks) {
+      // The callbacks are reconstructed when the data in the store changes
+      // We need to register the updated callbacks or the data they use will be stale
+      try {
+        serviceModule.registerCallbacks(serviceModuleCallbacks);
+      } catch {
+        /**
+         * Registration can error when the user switches between preview and debug mode.
+         * Registration is attempted with the old service, which has been destroyed.
+         * It then succeeds with the new one.
+         * @todo can we avoid this error?
+         */
+      }
+    }
+  }, [serviceModule, serviceModuleCallbacks]);
+
```

Comment:
@nathggns I think we can get rid of this and the re-registration below, right? The hooks should handle it for us.

@benwerner01 we recently removed the need for this, I believe – I appreciate you are following the existing pattern, which we should've cleaned up

Reply:
That's good to know, thank you. I've removed this `useEffect` in https://github.com/blockprotocol/blockprotocol/pull/966/commits/8d35f8e19029db8341683617c3639839f3696484. @nathggns let me know if I can remove the one for the `graphService`.

Reply:
All modules used via react now keep their callbacks up to date, so you can remove this